### ISSUE TITLE
windows ddp

### ISSUE THREAD

**Issue by LZLwoaini** [ultralytics/ultralytics#25653]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I want to know how to use DPP for training on Windows Server 2019. Thank you.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @LZLwoaini, thank you for your interest in Ultralytics 🚀! This is an automated response, and an Ultralytics engineer will also assist you soon. We recommend visiting the [Docs](https://docs.ultralytics.com/) for new users, where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and answers to common questions.

For your custom training ❓ question about DDP on Windows Server 2019, please provide as much information as possible, including:
- The exact command or Python script used
- Your `ultralytics` version and environment details
- Python and PyTorch versions
- GPU model, CUDA version, and number of GPUs
- Complete training logs and any error traceback
- A minimal dataset or configuration example, if possible
- Confirmation that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips)

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
It should work normally with lastest PyTorch. You just need to pass `device` argument and specify multiple devices 

Do you get any error?

**Comment by LZLwoaini** [comment]
> It should work normally with lastest PyTorch. You just need to pass `device` argument and specify multiple devices
> 
> Do you get any error?

<img width="1691" height="420" alt="Image" src="https://github.com/user-attachments/assets/012d2e7f-8d8c-4836-97a9-a308531a2d6e" />

<img width="461" height="885" alt="Image" src="https://github.com/user-attachments/assets/f3664491-2394-4a99-93d1-1ab558dd21a9" />                                                                                                                                                                                                                                   Yes, could you please take a look at what the problem is? Thank you.

**Comment by Y-T-G** [comment]
You can try upgrading to latest PyTorch. PyTorch 2.3.1 is old

**Comment by LZLwoaini** [comment]
> You can try upgrading to latest PyTorch. PyTorch 2.3.1 is old
I got an error after upgrading to the new version of torch. Do you know why?    

<img width="1705" height="864" alt="Image" src="https://github.com/user-attachments/assets/eb62a3f7-ba86-4dae-8cde-5c307f8f9e82" />

<img width="410" height="451" alt="Image" src="https://github.com/user-attachments/assets/f1452737-87e0-4920-b5a5-526a79b4c62d" />

**Comment by Y-T-G** [comment]
@LZLwoaini You can try adding this to your Python script at the top

```python
import os

os.environ["USE_LIBUV"] = "0"
os.environ["GLOO_USE_LIBUV"] = "0"
```

**Comment by Y-T-G** [comment]
@LZLwoaini Did it work?

**Comment by LZLwoaini** [comment]
> [@LZLwoaini](https://github.com/LZLwoaini) Did it work?

Sorry, I just saw your reply. Because of other work, I’ll probably try the YOLO training next Monday. Thanks, and if I succeed or run into new problems, I’ll ask for your help again.

**Comment by Y-T-G** [comment]
Hello @LZLwoaini 

Have you tried it?
